tests/codegen/mem-replace-simple-type.rs

```diff
@@ -1,6 +1,7 @@
 //@ compile-flags: -O -C no-prepopulate-passes
 //@ only-x86_64 (to not worry about usize differing)
-//@ ignore-debug: precondition checks make mem::replace not a candidate for MIR inlining
```

Comment:
Just to check: we have some lint or w/e that will fail CI if someone adds a test with `ignore-debug` as we merge this PR?

Reply:
compiletest (currently) have an allowlist of known directives, I've removed ignore-debug from that so compiletest will error if someone writes ignore-debug.

Reply:
See https://github.com/rust-lang/rust/pull/131913#discussion_r1807864577

Reply:
It will fail looking like (I know it's not super user friendly)

```
Testing stage1 compiletest suite=ui mode=ui (x86_64-pc-windows-msvc)
error: detected unknown compiletest test directive `only-debug` in X:\repos\rust\tests\ui\print_type_sizes\niche-filling.rs:13
errors encountered during EarlyProps parsing: X:\repos\rust\tests\ui\print_type_sizes\niche-filling.rs
thread 'main' panicked at src\tools\compiletest\src\header.rs:68:13:
errors encountered during EarlyProps parsing
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```